cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesExport.java

```diff
@@ -0,0 +1,408 @@
+                    String rtcId = twt.getAliasFromType(CGMES_PREFIX_ALIAS + CgmesNames.RATIO_TAP_CHANGER + i).orElseThrow(PowsyblException::new);
+                    writeSvTapStep(rtcId, leg.getRatioTapChanger().getTapPosition(), writer);
+                }
+                i++;
+            }
+        }
+    }
+
+    private static void writeSvTapStep(String tapChangerId, int tapPosition, XMLStreamWriter writer) throws XMLStreamException {
+        writer.writeStartElement(CIM_NAMESPACE, "SvTapStep");
+        writer.writeAttribute(RDF_NAMESPACE, ID, getUniqueId());
+        writer.writeStartElement(CIM_NAMESPACE, "SvTapStep.position");
+        writer.writeCharacters(is(tapPosition));
+        writer.writeEndElement();
+        writer.writeEmptyElement(CIM_NAMESPACE, "SvTapStep.TapChanger");
+        writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, "#" + tapChangerId);
+        writer.writeEndElement();
+    }
+
+    private static void writeStatus(Network network, XMLStreamWriter writer) {
+        // create SvStatus, iterate on Connectables, check Terminal status, add
+        // to SvStatus
+        network.getConnectableStream().forEach(c -> writeConnectableStatus((Connectable<?>) c, writer));
+
+        // RK: For dangling lines (boundaries), the AC Line Segment is considered in service if and only if it is connected on the network side.
+        // If it is disconnected on the boundary side, it might not appear on the SV file.
+    }
+
+    private static void writeConnectableStatus(Connectable<?> connectable, XMLStreamWriter writer) {
+        writeStatus(Boolean.toString(connectable.getTerminals().stream().anyMatch(Terminal::isConnected)), connectable.getId(), writer);
+    }
+
+    private static void writeStatus(String inService, String conductingEquipmentId, XMLStreamWriter writer) {
+        try {
+            writer.writeStartElement(CIM_NAMESPACE, "SvStatus");
+            writer.writeAttribute(RDF_NAMESPACE, ID, getUniqueId());
+            writer.writeStartElement(CIM_NAMESPACE, "SvStatus.inService");
+            writer.writeCharacters(inService);
+            writer.writeEndElement();
+            writer.writeEmptyElement(CIM_NAMESPACE, "SvStatus.ConductingEquipment");
+            writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, "#" + conductingEquipmentId);
+            writer.writeEndElement();
+        } catch (XMLStreamException e) {
+            throw new UncheckedXmlStreamException(e);
+        }
+    }
+
+    private static String getUniqueId() {
+        return UUID.randomUUID().toString();
+    }
+
+    // Avoid trailing zeros
+    private static final DecimalFormat DOUBLE_FORMAT = new DecimalFormat("0.##############", new DecimalFormatSymbols(Locale.US));
+
+    private static String fs(double value) {
+        return Double.isNaN(value) ? DOUBLE_FORMAT.format(0.0) : DOUBLE_FORMAT.format(value);
+    }
+
+    private static Complex complexVoltage(double r, double x, double g, double b,
```

Comment:
I am not sure to get why we introduce a new function here ? We already have a function to do that ? Or it is not the good place maybe.

Reply:
I put it in `CgmesExportUtil`